docs/DeveloperGuide.md

```diff
@@ -1,34 +1,245 @@
+SpendingList: Stores the expenditure of the user as individual entries
 
-## User Stories
+Storage: Reads data from, and writes data to, the hard disk
+
+Ui: Interacts with the user
+
+Command: Executes the user command or system-issued command
+
+Parser: Analyzes the user command
+
+**How the architecture components interact with each other** </br>
+
+The sequence diagram below shows how the components interact with each other when the user issues a general command.
+
+![image](https://user-images.githubusercontent.com/45732128/97583491-fa4cf800-1a31-11eb-98a5-3fe405076db2.png) </br>
+Figure 2: Interaction between architecture components
+
+
+### 3.2 Edit Feature
+`SpendingList` and `Ui` facilitate this feature. The Edit feature is able to edit the existing items in the spending list. 
+It implements the following operations:
+* `SpendingList#editItemDescription(index, description)` → updates the description of the item in the spending list
+* `SpendingList#editItemCategory(index, description)` → updates the category of the item in the spending list
+* `SpendingList#editItemAmount(index, amount)` → updates the amount of the item in the spending list
+* `Ui#printEdit(spendingList, index)` → prints the message to show the successful completion of the edition process
+
+Below shows an example of the usage:
+1. User executes `edit 1 bubble tea` command to edit the first item in the spending list to bubble tea 
+2. The `edit` command calls the `EditCommand#execute(spendingList, ui)` to complete the edition process
+
+Figure below shows the sequence diagram of `EditCommand` class.
+![Sequence Diagram of EditCommand class](images/EditCommand.png) </br>
+Figure 3: Sequence diagram of EditCommand class
+
+### 3.3 Convert Feature
+`SpendingList`, `Ui` and `Item` facilitate this feature. The Convert feature is able to convert the currency of the items 
+stored in the spending list. It implements the following operations:
+* `SpendingList#getSpendingList()` → retrieves the current spending list
+* `ConvertCommand#identifyCurrency(description)` → identifies the currency after conversion
+* `ConvertCommand#findExchangeRate()` → finds the corresponding exchange rate
+* `Item#editAmount(amount)` → updates the new amount after conversion
+* `Item#editSymbol(currency)` → updates the new currency after conversion
+* `SpendingList#updateSpendingList()` → updates the spending list after conversion 
+* `Ui#printConvertCurrency(outputCurrency)` → prints the message to show the successful completion of the conversion 
+process
+
+Below shows an example of the usage:
+1. User executes `convert SGD USD` command to convert the currency from SGD to USD
+2. The `convert` command calls the `ConvertCommand#execute(spendingList, ui)` to complete the conversion process
+
+Figure below shows the sequence diagram of `ConvertCommand` class.
+
+![Sequence Diagram of ConvertCommand class](images/ConvertCommand.png) </br>
```

Comment:
Is there a spelling mistake here whereby it should be `outputCurrency` instead of `outputCurrecny`?

![image](https://user-images.githubusercontent.com/59533849/97660512-c4992500-1aac-11eb-86df-0f87ca809f5d.png)